### PR TITLE
Cleanup ignores

### PR DESCRIPTION
**Pull request by provokateurin**

### CODE REVIEW

File: packages/neon_framework/test/option_test.dart [hunk 1]
```diff
@@ -282,7 +287,7 @@ void main() {
     });
 
     test('Disable', () {
-      final callback = MockCallbackFunction();
+      final callback = MockCallbackFunction<void>();
```
Comment (Leptopoda) [inline packages/neon_framework/test/option_test.dart:290]: it is similar here:
Do we want to test that it works with a void method or any arbitrary method? Yes we can use `dynamic` as a generic but I think just ignoring the rule makes more sense.

**Review by Leptopoda — COMMENTED** [review]
In the last commit I'd like to keep `inference_failure_on_instance_creation` disabled.
For the other lints :woman_shrugging:

File: packages/neon_framework/test/persistence_test.dart [hunk 2]
```diff
@@ -1,6 +1,3 @@
-// ignore_for_file: unnecessary_lambdas, inference_failure_on_instance_creation, strict_raw_type
-// ignore_for_file: inference_failure_on_collection_literal
```
[the remark this answers is not in the document]
Comment (provokateurin) [reply packages/neon_framework/test/persistence_test.dart:2]: I get your point, but I really dislike the file wide ignores. They can also hide unintended things, so only line based ignores should be used where necessary.

[the remark this answers is not in the document]
Comment (Leptopoda) [reply packages/neon_framework/test/persistence_test.dart:2]: what about only allowing file wide ignores if the ignored rule is `inference_failure_on_instance_creation` and it is inside a test file?
I don't think annotating each occurrence in a test with a line ignore for that rule.

Sounds like something a custom lint should do :)

**Comment by provokateurin** [comment]
I just rebased this and put the changes into the fixup commit. I hope this was all that was discussed?

**Review by Leptopoda — APPROVED** [review]
The only thing concerning me is the change of the `lerp` methods from static methods to factory constructors.
All flutter widgets I know of use a static method and I've never seen a factory constructor being used for them.
OTOH this is valid dart code and I can not imagine any reason why this shouldn't fly :woman_shrugging:

**Comment by provokateurin** [comment]
You're right, I only changed it because of the lint. In the end both ways are fine (but I can change it back if you want).

**Comment by Leptopoda** [comment]
I personally prefer the static methods but am fine with either way (if you prefer less ignores)

**Pull request merged**